var/spack/repos/builtin/packages/adios2/package.py

```diff
@@ -95,12 +100,18 @@ class Adios2(CMakePackage, CudaPackage):
     conflicts("%oneapi@:2022.1.0", when="+fortran")
 
     depends_on("cmake@3.12.0:", type="build")
-    depends_on("pkgconfig", type="build")
 
-    depends_on("libffi", when="+sst")  # optional in DILL
-    depends_on("libfabric@1.6.0:", when="+sst")  # optional in EVPath and SST
-    # depends_on('bison', when='+sst')     # optional in FFS, broken package
-    # depends_on('flex', when='+sst')      # optional in FFS, depends on BISON
+    for plat in ["linux", "darwin", "cray"]:
```

Comment:
can we use `platform` instead of `plat`, it makes it more clear at the expense of 4 more chars.

Reply:
`platform` is also the name of a python module and the platform designation in the spec so there's avenue for confusion and actual errors there as well. I feel like `plat` vs `platform` is a fairly minor distinction and FWIW we use `plat` in almost every case where this sentiment is expressed. I have no problem with changing the name here, but if we're doing this for clarity sake, we should consider all factors and consistency.

Reply:
Fair point with regard that it can also add confusion due to existing symbols being named as platform as well, although there is no naming overwrite.

I think there is still a net good of being a bit more verbose in this file since spack packages are often read by Spack users who can be very unfamiliar with both spack and ADIOS2, but for instance they might needs figure out which variant does what and or something else.

Alternatively, as we do in the C++/C world, how about underscoring the name of the variable such as `platform_`, it might look ugly in python but it will satisfy both requirements: clarity and unique name.